### ISSUE TITLE
Broken, Starting with Firefox 35.

### ISSUE THREAD

**Issue by ScottMichaud** [toaarnio/webcl-firefox#71]
Lesson 2 will not run, starting with Firefox 35.

On Firefox 35, Browser Console throws error:

 "Security wrapper denied access to property apply on privileged Javascript object. Support for exposing privileged objects to untrusted content via **exposedProps** is being gradually removed - use WebIDL bindings or Components.utils.cloneInto instead. Note that only the first denied property access from a given global object will be reported."

While Firefox 35 will still run Lesson 1, Firefox 36 does not. It throws the errors: 

"WEBCL ERROR: WebCLLoader#observe "content-document-global-created": ReferenceError: WebCLException is not defined" and
"ReferenceError: WebCLException is not defined"

**Comment by jurekd** [comment]
That's right. Firefox 34 is the last version when WebCL plugin works without problems.
After upgrading to Firefox 35 checking whether WebCL works (point 3 in Getting Started) returns me the message:
"Your browser supports WebCL in principle, but unfortunately the underlying operating system only supports OpenCL 1.0, whereas 1.1 would be required.  You may want to try updating your display driver, or installing a CPU-based OpenCL implementation (such as the Intel OpenCL SDK or the AMD APP SDK)."
Which is obviously not true, as I have two devices (Intel i7) and NVidia 540M running OpenCL 1.2 and OpenCL 1.1, respectively.
Please update the plugin to the latest version of Firefox.

**Comment by BitSlapper** [comment]
Does Nokia plan on fixing this or have they given up in WebCL?

**Comment by ScottMichaud** [comment]
Personally, I have maintained a Firefox 34 instance with a custom -no-remote WebCL profile. I mostly use an old version of Win64 Nightly from Mozilla's FTP (because there's a few experimental features that I might want but are disabled in Release).

https://ftp.mozilla.org/pub/mozilla.org/firefox/nightly/2014/08/2014-08-31-03-02-06-mozilla-central/
My choice is: firefox-34.0a1.en-US.win64-x86_64.zip

If you want to stick with release branches, then you can try 34.0.5.

https://ftp.mozilla.org/pub/mozilla.org/firefox/releases/34.0.5/

Really, the main reason why WebCL struggles is because browser vendors believe that "no-one" will develop for it. That's not without basis, given the lackluster adoption of OpenCL proper. OpenCL is also pivoting itself, preferring compiled LLVM-based bytecode kernels instead of full-source kernels that are compiled by the driver.

In either case, the best way to convince browsers to adopt it is software.

**Comment by BitSlapper** [comment]
Thank you for the links Scott!

I agree that the major browser developers aren't being given much reason to continue WebCL support right now.

I hope that changes soon. This is definitely an interesting project and I was impressed to see that Nokia even developed it in the first place.

Having to use Firefox version 34 when the latest is 35 isn't so bad. Hopefully it doesn't continue for to many versions without support though...

**Comment by ScottMichaud** [comment]
36 has been released for over a couple of weeks and it is now up to 36.0.1. Off topic, but you might want to get that checked out (if you're failing to update for one reason or another).

**Comment by BitSlapper** [comment]
Ahh, Ok thank you I'll check that out. I mainly use Chrome. I've only been using Firefox for some development.

**Comment by MayaLekova** [comment]
Unfortunately the extension still doesn't work in Firefox 36.0.4

**Comment by csak2040** [comment]
Someone knows, whether the work on the WebCL extension will be continoued?
